File: pc_builder_service/internal/domain/models/mother_board.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type MotherBoard struct {
	ID 								uuid.UUID 			`gorm:"type:uuid;primary_key"`
	Brand 							string 				`gorm:"type:varchar(255);not null"`
	Series 							string 				`gorm:"type:varchar(255);not null"`
	Socket 							string 				`gorm:"type:varchar(255);not null"`
	DDR 							string 				`gorm:"type:varchar(255);not null"`
	MemorySlots 					int32 				`gorm:"type:integer;not null"`
	MaxRam 							int32 				`gorm:"type:integer;not null"`
	MaxRamMemoryFrequencyMhz 		int32 				`gorm:"type:real;not null"`
	PciExpress 						int32 				`gorm:"column:pci_express;type:integer;not null"`
	M2Slots 						int32 				`gorm:"type:integer;not null"`
	Vrm 							int32 				`gorm:"type:integer;not null"`
	AvgPrice 						float32 			`gorm:"type:real;not null"`
	Score 							int32 				`gorm:"type:integer"`
	Img 							[]byte 				`gorm:"type:bytea"`
	CreatedAt 						time.Time 			`gorm:"type:timestamp;not null"`
	UpdatedAt 						time.Time 			`gorm:"type:timestamp"`
}

File: benchmark_service/src/main/java/com/buildpc/benchmark_service/entities/MotherBoard.java
package com.buildpc.benchmark_service.entities;

import com.google.type.DateTime;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.sql.Date;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "mother_boards")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MotherBoard {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private UUID id;
	private String brand;
	private String series;
	private String socket;
	private String ddr;
	private Integer memorySlots;

	@Column(name = "max_ram")
	private Integer maxRAM;
	private Float maxRamMemoryFrequencyMhz;
	private Integer pciExpress;
	private Integer M2Slots;
	private Integer vrm;
	private Float avgPrice;
	private Integer score;
	private byte[] img;

	@LastModifiedDate
	private LocalDateTime updatedAt;

	@CreatedDate
	private LocalDateTime createdAt;
}
